src/dash/issues/DashIssuesPage.tsx

```diff
@@ -0,0 +1,195 @@
+import { Table, Button, Modal, Form, Spinner } from "react-bootstrap";
+import { Link, useHistory } from "react-router-dom";
+import { useForm, ErrorMessage } from "react-hook-form";
+
+import "./DashIssuesPage.scss";
+import Field from "../../shared/form/Field";
+import ERRORS from "../../shared/errors";
+import { Issue } from "../../shared/types";
+
+const ISSUES_PER_VOLUME = 6;
+
+interface IssueCreateModalProps {
+  show: boolean;
+  setShow: (show: boolean) => void;
+  latestIssue: Issue;
+}
+
+interface IssueCreateFormVals {
+  issueNum: number;
+  volumeNum: number;
+}
+
+const IssueCreateModal: React.FC<IssueCreateModalProps> = (
+  props: IssueCreateModalProps
+) => {
+  const { handleSubmit, register, errors } = useForm<IssueCreateFormVals>({
+    defaultValues: {
+      volumeNum:
+        props.latestIssue.issue_num <= ISSUES_PER_VOLUME
+          ? props.latestIssue.volume_num + 1
+          : props.latestIssue.volume_num,
+      issueNum: (props.latestIssue.issue_num + 1) % ISSUES_PER_VOLUME,
+    },
+  });
+
+  const history = useHistory();
+
+  const [createIssue, ,] = useCreateIssue();
+
+  const [nonFieldErrors, setNonFieldErrors] = useState<string[]>([]);
+
+  const closeModal = () => {
+    setNonFieldErrors([]);
+    props.setShow(false);
+  };
+
+  const onSubmit = async (vals: IssueCreateFormVals) => {
+    const resp = await createIssue({
+      issue_num: vals.issueNum,
+      volume_num: vals.volumeNum,
+    });
+    if (resp.success) {
+      setNonFieldErrors([]);
+      history.push(`issues/${resp.data.id}`);
+    } else {
+      setNonFieldErrors(resp.error.non_field_errors || []);
+    }
+  };
+
+  return (
+    <Modal show={props.show} onHide={closeModal}>
+      <Modal.Header closeButton>Create New Issue</Modal.Header>
+      <Modal.Body>
+        <Form.Label>Issue Name:</Form.Label>
+        <Form id="createIssueForm" onSubmit={handleSubmit(onSubmit)} inline>
+          <Form.Label>v</Form.Label>
+          <Field
+            name="volumeNum"
+            className="volume-num-input"
+            type="text"
+            maxLength={3}
+            ref={register({
+              required: true,
+              pattern: /[0-9]{3}/,
+              min: 0,
+              maxLength: 3,
+            })}
+            errors={errors}
+            hideErrorMessage
+          />
+          <Form.Label>i</Form.Label>
+          <Field
+            name="issueNum"
+            className="issue-num-input"
+            type="text"
+            maxLength={1}
+            ref={register({
+              required: true,
+              pattern: /[0-9]{1}/,
+              min: 0,
+              maxLength: 1,
+            })}
+            errors={errors}
+            hideErrorMessage
+          />
+        </Form>
+        {nonFieldErrors?.map((error, idx) => (
+          <small key={idx} className="invalid-feedback d-block">
+            {ERRORS[error]}
+          </small>
+        ))}
```

Comment:
Can we componentize this map? I like it.

Reply:
(sidenote: if we do componentize this, it would probably be better to use the error as the key instead of the index, since errors should not be duplicated)

Reply:
A `<NonFieldError>` or `<SubmissionError>` component makes sense.